app/models/manageiq/providers/lenovo/physical_infra_manager/parsers/parser.rb

```diff
@@ -35,7 +35,17 @@ def parse_physical_server(node)
       result[:health_state]               = dictionary::HEALTH_STATE_MAP[node.cmmHealthState.nil? ? node.cmmHealthState : node.cmmHealthState.downcase]
       result[:host]                       = get_host_relationship(node.serialNumber)
       result[:location_led_state]         = find_loc_led_state(node.leds)
-      result[:computer_system][:hardware] = get_hardwares(node)
+      result[:computer_system][:hardware] = get_physical_server_hardwares(node)
+
+      return node.uuid, result
+    end
+
+    def parse_switch(node, ems_id = nil)
+      result = parse(node, dictionary::SWITCH)
+      result[:hardware] = get_switch_hardwares(node)
+
+      # Keep track of the ems where this switch is in.
+      result[:ems_id] = ems_id unless ems_id.nil?
```

Comment:
The `.nil?` is not needed here, you can just do `result[:ems_id] = ems_id if ems_id`